moto/ec2/models.py

```diff
@@ -3511,6 +3524,295 @@ def modify_subnet_attribute(self, subnet_id, attr_name, attr_value):
+
+        * deliver-log-status
+        * log-destination-type
+        * flow-log-id
+        * log-group-name
+        * resource-id
+        * traffic-type
+        * tag:key=value
+        * tag-key
+
+        Taken from: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeFlowLogs.html
+        """
+        if filter_name == "resource-id":
+            return self.resource_id
+        elif filter_name == "traffic-type":
+            return self.traffic_type
+        elif filter_name == "log-destination-type":
+            return self.log_destination_type
+        elif filter_name == "flow-log-id":
+            return self.id
+        elif filter_name == "log-group-name":
+            return self.log_group_name
+        elif filter_name == "deliver-log-status":
+            return "SUCCESS"
+        else:
+            return super(FlowLogs, self).get_filter_value(
+                filter_name, "DescribeFlowLogs"
+            )
+
+
+class FlowLogsBackend(object):
+    def __init__(self):
+        self.flow_logs = defaultdict(dict)
+        super(FlowLogsBackend, self).__init__()
+
+    def _validate_request(
+        self,
+        log_group_name,
+        log_destination,
+        log_destination_type,
+        max_aggregation_interval,
+        deliver_logs_permission_arn,
+    ):
+        if log_group_name is None and log_destination is None:
+            raise InvalidDependantParameterError(
+                "LogDestination", "LogGroupName", "not provided",
+            )
+
+        if log_destination_type == "s3":
+            if log_group_name is not None:
+                raise InvalidDependantParameterTypeError(
+                    "LogDestination", "cloud-watch-logs", "LogGroupName",
+                )
+        elif log_destination_type == "cloud-watch-logs":
+            if deliver_logs_permission_arn is None:
+                raise InvalidDependantParameterError(
+                    "DeliverLogsPermissionArn",
+                    "LogDestinationType",
+                    "cloud-watch-logs",
+                )
+
+        if max_aggregation_interval not in ["60", "600"]:
+            raise InvalidAggregationIntervalParameterError(
+                "Flow Log Max Aggregation Interval"
+            )
+
+    def create_flow_logs(
+        self,
+        resource_type,
+        resource_ids,
+        traffic_type,
+        deliver_logs_permission_arn,
+        log_destination_type,
+        log_destination,
+        log_group_name,
+        log_format,
+        max_aggregation_interval,
+    ):
+        # Guess it's best to put it here due to possible
+        # lack of them in the CloudFormation template
+        max_aggregation_interval = (
+            "600" if max_aggregation_interval is None else max_aggregation_interval
+        )
+        log_destination_type = (
+            "cloud-watch-logs" if log_destination_type is None else log_destination_type
```

Comment:
Can we have a test where LogDestinationType is explicitly set to `cloud-watch-logs`? At the moment there are only tests for s3, and tests where the param is not provided (None)

Reply:
Sure, I'll add this